internal/provider/provider.go

```diff
@@ -1936,11 +1936,20 @@ func assumeRoleSchema() *schema.Schema {
 		MaxItems: 1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"duration": {
+					Type:          schema.TypeString,
+					Optional:      true,
+					Description:   "The duration of the role session, e.g. 1h. Valid time units are ns, us (or µs), ms, s, h, or m.",
+					ValidateFunc:  verify.ValidTimeDuration,
```

Comment:
We could also check for the minimum value of 15 minutes and the absolute maximum of 12 hours

Reply:
Test output after updates:
```
--- PASS: TestValidAssumeRoleDuration (0.00s)
```